optimum/intel/openvino/modeling.py

```diff
@@ -996,3 +997,23 @@ def forward(self, input_ids, pixel_values, attention_mask: Optional[torch.Tensor
             text_embeds=text_embeds,
             image_embeds=image_embeds,
         )
+
+    @classmethod
+    def from_pretrained(cls, model_id, *args, **kwargs):
+        load_in_8bit = kwargs.pop("load_in_8bit", False)
+        quantization_config = kwargs.pop("quantization_config", None)
+        quantization_config = cls._prepare_quantization_config(quantization_config, load_in_8bit)
+        is_data_aware_quantization = quantization_config is not None and quantization_config.dataset is not None
+
+        model = super().from_pretrained(
+            model_id, *args, quantization_config=None if is_data_aware_quantization else quantization_config, **kwargs
+        )
+
+        if is_data_aware_quantization:
+            from optimum.intel import OVQuantizer
+
+            quantization_config_copy = copy.deepcopy(quantization_config)
+            quantization_config_copy.processor = quantization_config.processor or model_id
+            ov_quantizer = OVQuantizer(model)
+            ov_quantizer.quantize(ov_config=OVConfig(quantization_config=quantization_config_copy))
+        return model
```

Comment:
shouldn't this be done in OVBaseModel ? https://github.com/huggingface/optimum-intel/blob/1d8cdcaf241b599369bc3d2e57a0be471db9d5ed/optimum/intel/openvino/modeling_base.py#L399

Reply:
Yes, indeed. Removed.